lib/backends/webgl/ops/conv.ts

```diff
@@ -169,18 +166,18 @@ export class WebGLConv extends Conv {
     const xshape = inputs[0].dims.slice();
     const kshape = inputs[1].dims.slice();
     const adjustedKernelShape = [kshape[0], Math.ceil((xshape[1] * kshape[2] * kshape[3]) / 4)];
-    const kLayout = inferenceHandler.createBasicTextureLayout(
+    const kLayout = inferenceHandler.createTextureLayoutFromShape(
         adjustedKernelShape, 4, [adjustedKernelShape[0], adjustedKernelShape[1] * 4], {breakAxis: 1});
 
     let bLayout: TextureLayout|undefined;
     const rank = outputShape.length;
 
     const inputLayouts = [im2colLayout, kLayout];
     if (inputs.length === 3) {
-      bLayout = inferenceHandler.createBasicTextureLayout(inputs[2].dims.slice());
+      bLayout = inferenceHandler.createTextureLayoutFromShape(inputs[2].dims.slice());
```

Comment:
There seem to be a lot of unnecessary slicing (copying) of `dims` in general - as long as the methodd accepts const dims and guarantees that any given dims will not be mutated - can we avoid creating copies unnecessarily ? Granted -it should be only adding to the warm up overhead, but it still kind of stands out...

Reply:
thanks for figuring out this. actually there are quite some unnecessary calls to `Array.prototype.slice` in a lot of places in the code base.

I think it worth another separated change to optimize that

Reply:
Oh yeah, I seem to have this comment already - too many `slice` calls :)